src/zimcheck/checks.cpp

```diff
@@ -487,15 +488,26 @@ void ArticleChecker::detect_redundant_articles()
         progress.report();
         auto l = it.second;
         while ( !l.empty() ) {
-            const auto e1 = archive.getEntryByPath(l.front());
+            // The way we have constructed `l`, e1 MUST BE an item
+            const auto e1 = archive.getEntryByPath(l.front()).getItem();
             l.pop_front();
+            const auto e1_cluster_idx = e1.getClusterIndex();
+            const auto e1_blob_idx = e1.getBlobIndex();
             if ( !l.empty() ) {
-                // The way we have constructed `l`, e1 MUST BE an item
-                const std::string s1 = e1.getItem().getData();
+                std::optional<std::string> s1;
                 decltype(l) articlesDifferentFromE1;
                 for(auto other : l) {
-                    auto e2 = archive.getEntryByPath(other);
-                    std::string s2 = e2.getItem().getData();
+                    // The way we have constructed `l`, e2 MUST BE an item
+                    const auto e2 = archive.getEntryByPath(other).getItem();
+                    const auto e2_cluster_idx = e2.getClusterIndex();
+                    const auto e2_blob_idx = e2.getBlobIndex();
+                    if (e1_cluster_idx == e2_cluster_idx && e1_blob_idx == e2_blob_idx) {
```

Comment:
I'd like to have here a self-explaining `if ( areAliases(e1, e2) )`  and less clutter in this function due to the extra variables `e?_cluster_idx` and `e?_blob_idx`.